api/maps/AddMarkers.tsx

```diff
@@ -129,6 +130,50 @@ export default function AddMarker({
     return setClusterer;
   }, [map]);
 
+  const markerMap = useRef<Map<number, google.maps.Marker>>(new Map());
```

Comment:
This creates `markerMap` which is like a hashmap that maps the project ids to its corresponding marker (google.maps.Marker). `useRef` is just a hook that makes it so that the component doesn't re-render every time we update the map. Instead, `useRef` creates a mutable object (markerMap.current) that exists across re-renders of the component.